lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
IMO, we should be upfront with the behaviour of `Logger` and clearly document that calls to `Logger` macros with a level lower than `Logger.level` will be completely purged from the AST, meaning whatever is passed to them is not computed (not even `fn`s, as it would bring confusion IMO). @lexmag can you think of any cases where you may do something like 

```elixir
Logger.debug "Done: #{cleanup()}"
```

instead of

```elixir
cleanup_res = cleanup()
Logger.debug "Done: #{cleanup_res}"
```

? I feel like doing the former would just be bad practice. Wdyt?